engine.py

```diff
@@ -3,306 +3,319 @@
+            and appends them to a list
+
+            input parameter(s):
+            r     --> starting row (int)
+            c     --> starting colum (int)
+            moves --> possible moves container (list)
+
+            return parameter(s):
+            None
+        """
+
+        ## FIX
+        if self.light_to_move: # if it's light's turn to move
+
+            for i in range(len(self.board)):
+                for j in range(len(self.board[i])):
+                    #if self.board[i][j] == "  " or self.board[i][j][1] == "d": # if square is empty or square has opponent's piece
+                    #	moves.append(Move((r, c), (i, j), self.board)) # create a move object and append to list
+                    if ((i == r-2) and (j == c) and (r==6) and (self.board[i][j] == "  ")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+                    if ((i == r-1) and (j == c) and (self.board[i][j] == "  ")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+                    if ((i == r-1) and (j == c+1 or j==c-1) and (self.board[i][j][1] == "d")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+
+        ##FIX
+        else: # if it's dark's turn to move
+
+            for i in range(len(self.board)):
+                for j in range(len(self.board[i])):
+                    #if self.board[i][j] == "  " or self.board[i][j][1] == "l": # if square is empty or square has opponent's piece
+                    #	moves.append(Move((r, c), (i, j), (self.board))) # create a move object and append to moves
+                    if ((i == r+2) and (j == c) and (r==1) and (self.board[i][j] == "  ")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+                    if ((i == r+1) and (j == c) and (self.board[i][j] == "  ")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+                    if ((i == r+1) and (j == c+1 or j==c-1) and (self.board[i][j][1] == "l")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+
+    def get_bishop_moves(self, r, c, moves):
+
+         """
+            calculates all possible bishop moves for a given colour (light or dark)
+            and appends them to a list
+
+            input parameters:
+            r     --> starting row (int)
+            c     --> starting column (int)
+            moves --> posiible moves container (list)
+
+            return parameter(s):
+            None
+         """
+
+         ##TODO
+         pass
+
+
+    def get_knight_moves(self, r, c, moves):
+         ##TODO
+         pass
+
+    
+    def get_king_moves(self, r, c, moves):
+         ##TODO
+         pass
+
+
+
+    def get_rook_moves(self, r, c, moves):
+         direction = (-1, 1) # possible direction
+         
+         if self.light_to_move:
+            
+
+            # if it's light's turn to move
```

Comment:
move comment to line 123 or above it

Reply:
Move this comment to the same line with if self.light_to_move or to the line above it